### ISSUE TITLE
Inner loop optimization is inconsistent.

### ISSUE THREAD

**Issue by Kaushalya** [Kaushalya/gnn-meta-attack-pytorch#1]
Inner loop of training inside GNNAttack.perturb_adj() results in inconsistent behaviour of loss.
Eg: 
### Perturbation: 1  
epoch:0 train-loss = 1.2192912101745605  
epoch:1 train-loss = **4.535911560058594**
epoch:2 train-loss = 0.9778962731361389  
epoch:3 train-loss = 0.39638981223106384  
epoch:4 train-loss = 2.0525920391082764
